rust/connlib/clients/apple/build-rust.sh

```diff
@@ -6,56 +6,71 @@
 
 set -e
 
+# Sanitize the environment to prevent Xcode's shenanigans from leaking
+# into our highly evolved Rust-based build system.
+for var in $(env | awk -F= '{print $1}'); do
+  if [[ "$var" != "HOME" ]] \
+  && [[ "$var" != "USER" ]] \
+  && [[ "$var" != "LOGNAME" ]] \
+  && [[ "$var" != "TERM" ]] \
+  && [[ "$var" != "PWD" ]] \
+  && [[ "$var" != "SHELL" ]] \
+  && [[ "$var" != "TMPDIR" ]] \
+  && [[ "$var" != "XPC_FLAGS" ]] \
+  && [[ "$var" != "XPC_SERVICE_NAME" ]] \
+  && [[ "$var" != "PLATFORM_NAME" ]] \
+  && [[ "$var" != "CONFIGURATION" ]] \
+  && [[ "$var" != "NATIVE_ARCH" ]] \
+  && [[ "$var" != "ONLY_ACTIVE_ARCH" ]] \
+  && [[ "$var" != "ARCHS" ]] \
+  && [[ "$var" != "SDKROOT" ]] \
+  && [[ "$var" != "OBJROOT" ]] \
+  && [[ "$var" != "SYMROOT" ]] \
+  && [[ "$var" != "SRCROOT" ]] \
+  && [[ "$var" != "TARGETED_DEVICE_FAMILY" ]] \
+  && [[ "$var" != "CONNLIB_TARGET_DIR" ]]; then
+  unset $var
+  fi
+done
+
+# Use pristine path; the PATH from Xcode is polluted with stuff we don't want which can
+# confuse rustc.
+export PATH="/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/homebrew/bin:$HOME/.cargo/bin"
```

Comment:
What was happening was these were polluting the environment causing `rustc` to link against the wrong libraries. After much trial and error, it seems that it was a combination of the env vars Xcode sent along with the `PATH` it sets that broke things.

See here for the difference between my local shell and a failing Xcode 15 build run:

https://gist.github.com/jamilbk/3b19f9f1a724344ea60da9277c7750a0/revisions